src/service/extension-script.js

```diff
@@ -159,25 +160,45 @@ export function createExtensionScript(win, extensionId, version) {
  * extension ID, if it exists. Otherwise, returns null.
  * @param {!Window} win
  * @param {string} extensionId
+ * @param {string} version
+ * @param {boolean} latest
  * @param {boolean=} includeInserted If true, includes script elements that
  *   are inserted by the runtime dynamically. Default is true.
  * @return {!Array<!Element>}
  */
-export function getExtensionScripts(win, extensionId, includeInserted = true) {
+export function getExtensionScripts(
+  win,
+  extensionId,
+  version,
+  latest,
```

Comment:
I don't think we really  need `version` or `latest` here for any of the uses, and it just causes more confusion. Eg, `isExtensionScriptRequired_`, which won't properly catch `latest` version.

Reply:
This case is specifically important for `reloadExtension`. What happens is this: the extension script arrives and we see that it's from a wrong RTV, so we re-request it. We need to do two things: (a) mark the old scripts as "skipped", and (b) insert a script for the fresh RTV.  The (b) is easy since we now have the version, so we just construct the script as `/rtv/${RTV/${ext}-${version}.js`.

(a) is harder however. We don't know how the extension was loaded. It could be done either with `${ext}-${version}.js` or `${ext}-latest.js`. So we need to find either option. Or both. Previously we've updated every single script for the same extension. But I don't think that was right, since some of them could have been of different versions. E.g. a fact that a wrong RTV of `amp-mustache-0.1` has been loaded, doesn't mean that the `amp-mustache-0.2` was also loaded with a wrong RTV.

From what I can tell, this is the only use case that needs to be aware of the `latest` flag. However, all other use cases need at least `version`. E.g. if a parent page has `amp-mustache:0.1` and FIE is asking for `amp-mustache:0.2`, we need to be able to check whether the `0.2` version specifically is already on the page.